Unosquare.FFME/Primitives/WorkerBase.cs

```diff
@@ -0,0 +1,276 @@
+﻿namespace Unosquare.FFME.Primitives
+{
+    using System;
+    using System.Diagnostics;
+    using System.Runtime.CompilerServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal abstract class WorkerBase : IWorker
+    {
+        private readonly object SyncLock = new object();
+        private readonly Stopwatch CycleClock = new Stopwatch();
+        private readonly ManualResetEventSlim WantedStateCompleted = new ManualResetEventSlim(true);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Implement 'IDisposable' in this class and use the 'Dispose' method to call 'Dispose' on 'TokenSource', 'WantedStateCompleted'.](https://app.codacy.com/app/UnosquareLabs/ffmediaelement/pullRequest?prid=4726833)